aten/src/ATen/native/ComplexHelper.cpp

```diff
@@ -0,0 +1,54 @@
+#include <ATen/native/ComplexHelper.h>
+
+#include <ATen/native/TensorIterator.h>
+
+namespace at {
+namespace native {
+
+DEFINE_DISPATCH(complex_stub);
+DEFINE_DISPATCH(complex_polar_stub);
+
+Tensor& complex_out(Tensor& result, const Tensor& real, const Tensor& imag) {
+  TORCH_CHECK(real.scalar_type() == imag.scalar_type(),
+              "Expected object of scalar type ", real.scalar_type(), " but got "
+              "scalar type ", imag.scalar_type(), " for argument 'imag'");
+  // Sort of hacky, but necessary so that input dtypes don't get promoted.
```

Comment:
there should be a check for shapes
```
TORCH_CHECK(real.sizes() == imag.sizes(),
      "complex expects real and imag tensors to have equal size, but got ", real.sizes() ,
      " and ", imag.sizes());
```

Reply:
and then we should do: `result.resize_(real.sizes());` which resizes the result if required

Reply:
Hmm right now when I do `torch.complex(x, y)` and they are of different sizes, I get for example `The size of tensor a (4) must match the size of tensor b (2) at non-singleton dimension 1`. This is the exact same as if I called `torch.add(x, y)`. I'm guessing it's handled by TensorIterator, which I think handles the resizing too.